apis/r/R/SOMANDArrayBase.R

```diff
@@ -135,6 +135,21 @@ SOMANDArrayBase <- R6::R6Class(
       coords
     },
 
+    #  @description Converts a vector of ints into a vector of int64 in a format
+    #  acceptable for libtiledbsoma
+
+    .convert_shape_argument = function(newshape) {
+
+      ## ensure newshape is a vector of int64
+      stopifnot("'newshape' must be a vector of integers" = is.vector(newshape));
+                # XXX "'newshape' must be a vector of integers" = all(vapply_lgl(newshape, is.integer)),
+
+      ## convert integer to integer64 to match dimension type
+      newshape <- lapply(newshape, function(x) if (inherits(x, "integer")) bit64::as.integer64(x) else x)
+
+      as.vector(newshape)
+    },
```

Comment:
This function can be strengthened and simplified by:
 - having stronger checking for integerish values
 - making use of vectorized `bit64::as.integer64()`
 - making use of method dispatch of `bit64::as.integer64()` to handle cases where `new_shape` is integer, numeric, or integer64

Also uses `new_shape` to be consistent with style
```suggestion
    .convert_shape_argument = function(new_shape) {
      # ensure new_shape is an integerish vector
      stopifnot(
        "'new_shape' must be an integerish vector" = rlang::is_integerish(new_shape, n = self$ndim(), finite = TRUE) ||
          (bit64::is.integer64(new_shape) && length(new_shape) == self$ndim() && is.finite(new_shape))
      )
      
      # convert integer to integer64 to match dimension type
      return(bit64::as.integer64(new_shape))
    },
```

However, I don't see any place where this is used. Will this be used? If so, why not just call `bit64::as.integer64()` where needed? I don't see a case where this is actually needed

Reply:
Thanks @mojaveazure -- code push coming up soon! Multi-tasking this morning. :)